// Protocol buffer to stream text data.
syntax = "proto3";

package multiscope.scalar;

import "tree.proto";

option go_package = "multiscope/protos/scalar_go_proto";

message Writer {
  // Path in the Multiscope tree.
  multiscope.NodePath path = 1;
}

// RPC API

// Request to create a new writer in the tree.
message NewWriterRequest {
  multiscope.NodePath path = 1;
}

// Response after creating a new writer in the tree.
message NewWriterResponse {
  Writer writer = 1;
}

// Request to write a scalars data.
message WriteRequest {
  Writer writer = 1;
  map<string, double> label_to_value = 2;
}

message WriteResponse {}

service Scalars {
  // Create a new scalars writer node in Multiscope.
  rpc NewWriter(NewWriterRequest) returns (NewWriterResponse) {}

  // Write scalars to Multiscope.
  rpc Write(WriteRequest) returns (WriteResponse) {}
}
